graphql_compiler/compiler/ir_lowering_common.py

```diff
@@ -225,32 +223,41 @@ def extract_optional_location_root_info(ir_blocks):
     """
     complex_optional_roots = []
     location_to_optional_root = dict()
-    in_optional_location = None
+    in_optional_root_location = None
     encountered_traverse_within_optional = False
 
-    for previous_block, current_block in pairwise(ir_blocks):
+    preceding_location = None
+    for current_block in ir_blocks:
         if isinstance(current_block, Traverse) and current_block.optional:
-            if in_optional_location is not None:
-                raise AssertionError(u'in_optional_location was not None at an optional traverse: '
-                                     u'{} {}'.format(current_block, ir_blocks))
-            if not isinstance(previous_block, MarkLocation):
-                raise AssertionError(u'No MarkLocation found before an optional traverse: '
-                                     u'{} {}'.format(current_block, ir_blocks))
-            in_optional_location = previous_block.location
-        elif in_optional_location is not None and isinstance(current_block, (Traverse, Recurse)):
+            if in_optional_root_location is not None:
+                raise AssertionError(u'in_optional_root_location was not None at an optional '
+                                     u'traverse: {} {}'.format(current_block, ir_blocks))
```

Comment:
This is my unfortunate code, but can you make `traverse` -> `Traverse` everywhere in the errors and docstrings since it's a type?